superset/assets/cypress/integration/dashboard/dashboard_controls.js

```diff
@@ -0,0 +1,51 @@
+describe('dashboard top-level controls', () => {
+  let sliceIds = [];
+  let dashboard = {};
+
+  beforeEach(() => {
+    cy.server();
+    cy.login();
+
+    cy.visit('/superset/dashboard/world_health');
+
+    cy.get('#app').then((data) => {
+      const bootstrapData = JSON.parse(data[0].dataset.bootstrap);
+      dashboard = bootstrapData.dashboard_data;
+      sliceIds = dashboard.slices.map(slice => (slice.slice_id));
+
+      // make sure dashboard is not favorite
+      cy.request(`/superset/favstar/Dashboard/${dashboard.id}/unselect/`);
```

Comment:
(just a thought) you could also toggle below, noting the start/end class. that would avoid coupling / hard-coding the endpoint.

Reply:
fixed. also move a few shared endpoints to a shared constants file.